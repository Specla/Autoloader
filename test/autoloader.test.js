/* eslint-env jest */

import path from 'path'
import autoloader from '../src/autoloader'

test('Should include the correct folders and create a namespace module', () => {
  const modules = autoloader(path.resolve(__dirname, './modules'))
  expect(modules).toMatchSnapshot()
})

test('Should ignore the folder submodules and file.js', () => {
  const modules = autoloader(path.resolve(__dirname, './modules'), {
    ignore: ['submodules', 'file.js']
  })

  expect(modules.submodules).toBe(undefined)
  expect(modules.file).toBe(undefined)
})

test('Should only include json files', () => {
  const modules = autoloader(path.resolve(__dirname, './modules'), {
    include: ['json']
  })

  expect(modules.submodules.data.data.samples).toMatchSnapshot()
  expect(modules.file).toBe(undefined)
})

test('Should work width ES module', () => {
  const modules = autoloader(path.resolve(__dirname, './modules'))

  expect(modules.submodules.esModule.hello).toBe('world')
  expect(modules.submodules.esModule).toMatchSnapshot()
})

test('ES Module should export primitive', () => {
  const modules = autoloader(path.resolve(__dirname, './modules'))
  expect(modules.submodules.esModulePrimitive).toBe(true)
})

test(`Should throw an exception if the path doesn't exist`, () => {
  expect(() => autoloader('./does-not-exist')).toThrowErrorMatchingSnapshot()
})

test(`Should throw an exection if the first argument isn't a string`, () => {
  expect(() => autoloader({})).toThrowErrorMatchingSnapshot()
})
