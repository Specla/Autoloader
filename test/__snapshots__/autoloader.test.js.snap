// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should include the correct folders and create a namespace module 1`] = `
Object {
  "file": Object {
    "key": "value",
  },
  "other-modules": Object {
    "test": Array [
      "other module",
    ],
  },
  "submodules": Object {
    "data": Object {
      "data": Object {
        "samples": Array [
          "test",
          "hello",
        ],
      },
    },
    "module": [Function],
  },
}
`;

exports[`Should only include json files 1`] = `
Array [
  "test",
  "hello",
]
`;

exports[`Should throw an exception if the path doesn't exist 1`] = `"The path ./does-not-exist does not exists"`;

exports[`Should throw an exection if the first argument isn't a string 1`] = `"First argument should be a string"`;

exports[`Should work width ES module 1`] = `[Function]`;
